streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -837,6 +833,10 @@ void runOnce() {
             }
         }
 
+        // we can always let changelog reader try restoring in order to initialize the changelogs;
+        // if there's no active restoring or standby updating it would not try to fetch any data
+        changelogReader.restore();
```

Comment:
This isn't a necessary part of the fix, but I feel it makes more sense to do this here: the restore consumer's can only make progress on tasks that have been initialized, which only happens in `taskManager.tryToCompleteRestoration`

Reply:
I agree. Originally I put it in front intentionally since `initializeIfNeeded` is called at task creation not here, and we would only try to initialize the changelogs in `restore` and hence only check if the restoration can be completed.

Now that we've delayed the `initializeIfNeeded` in `tryToComplete` we should reorder the restore call as well.